packages/types/Result.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+type ResultBase = {
+  readonly _tag: 'Result';
+};
+
+type NewErrorResultParams = {
+  message: string;
+  statusCode?: number;
+  name?: string;
+};
+
+type ErrorResult = ResultBase & {
+  status: 'error';
+  message: string;
+  statusCode: number;
+  name: string;
+};
+
+export const err = ({
+  message,
+  statusCode = 500,
+  name = 'Error',
+}: NewErrorResultParams): ErrorResult => ({
+  _tag: 'Result',
+  status: 'error',
+  message,
+  statusCode,
+  name,
+});
+
+type SuccessResult<TData> = ResultBase & {
+  status: 'success';
+  data: TData;
+  statusCode: number;
+};
+
+type NewSuccessResultParms<TData> = {
+  data: TData;
+  statusCode?: number;
+};
+
+export const ok = <TData>({
```

Comment:
I think these should follow [the naming convention for object creators](https://github.com/techmatters/aselo-coding-standards/blob/main/TypeScript.md#61-object-creator-function-naming) of having the verb prefix `new`: ex `newOk` so it is clear at a glance that all the function does is return a formatted object, it doesn't actually do any real work.